src/attr/_make.py

```diff
@@ -622,8 +625,20 @@ def slots_setstate(self, state):
                 __bound_setattr(_hash_cache_field, None)
 
         # slots and frozen require __getstate__/__setstate__ to work
-        cd["__getstate__"] = slots_getstate
-        cd["__setstate__"] = slots_setstate
+        if hasattr(self._cls, "__getstate__"):
+            if not hasattr(self._cls, "__setstate__"):
+                raise ValueError(
+                    "__setstate__ must be implemented when __getstate__ is."
+                )
+        else:
+            # TODO look closer! Some classes (like Exception) implements
```

Comment:
Actually one shouldn't override these in case only `__setstate__` is implemented either, this fails (on master as well!):
```
@attr.s(slots=True)
class MyException(ValueError):
    a = attr.ib()

def test_pickle_slotted_exception():
    me = MyException('a')
    me_new = pickle.loads(pickle.dumps(me))
    assert MyException('a') == me_new
```

Reply:
```
self = <pickle.Unpickler instance at 0x108ea4680>

    def load_reduce(self):
        stack = self.stack
        args = stack.pop()
        func = stack[-1]
>       value = func(*args)
E       TypeError: __init__() takes exactly 2 arguments (1 given)
```

Reply:
(might be a slightly different problem though?)